packages/fe-container/src/hooks/useAppData/useAppData.ts

```diff
@@ -27,13 +28,34 @@ const navigateToFirstPlugin = (configuration: Configuration) => {
   }
 }
 
+const setCssProperty = (propertyName: string, propertyValue: string | undefined) => {
+  const propertyColor = tinycolor(propertyValue)
+  const propertyToApply = propertyColor.isValid() ? propertyColor.toString() : getComputedStyle(document.documentElement).getPropertyValue(propertyName)
+  document.documentElement.style.setProperty(propertyName, propertyToApply)
+}
+
+const applyColorAlpha = (propertyValue: string | undefined, alphaValue: number) => {
+  const propertyColor = tinycolor(propertyValue)
+  if (propertyColor.isValid()) {
+    propertyColor.setAlpha(alphaValue)
+  }
+  return propertyColor.toRgbString()
+}
+
+const manageTheming = (configuration: Configuration) => {
+  document.title = configuration.theming?.header?.pageTitle || document.title
+  const primaryColor = configuration.theming?.variables.primaryColor
+  setCssProperty(COLORS.primaryColor, primaryColor)
+  setCssProperty(COLORS.menuEntrySelectedBackgroundColor, applyColorAlpha(primaryColor, 0.2))
+}
+
```

Comment:
We could move these three functions into a separate file specifically for themes management. So we leave `useAppData` cleaner and we can better unit test the themes releted functions.